dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowAction.java

```diff
@@ -1,155 +1,273 @@
+import java.io.Serializable;
+import java.util.*;
 
-
+/**
+ *
+ *
+ * @author root
+ * @version 1.x
+ * @since Mar 22, 2012
+ */
 public class WorkflowAction implements Permissionable, Serializable{
+
+	// requires lock options:
+	public static final String LOCKED   			  = "locked";
+	public static final String UNLOCKED 			  = "unlocked";
+	public static final String LOCKED_OR_UNLOCKED     = "both";
+
+	private final static Set<String> REQUIRES_LOCK_OPTION_SET =
+			Collections.unmodifiableSet(new HashSet<>(Arrays.asList(LOCKED, UNLOCKED, LOCKED_OR_UNLOCKED)));
+
+	private static final long serialVersionUID = 1L;
+
+	private String id;
+	private String name;
+	private String stepId;
+	private String schemeId;
+	private String condition;
+	private String nextStep;
+	private String nextAssign;
+	private String icon;
+	private boolean roleHierarchyForAssign;
+	private boolean requiresCheckout;
+	private String  requiresCheckoutOption;
+	private boolean assignable;
+	private boolean commentable;
+	private int order;
+
+	public WorkflowAction() {
+	}
+
 	@JsonIgnore
 	public String getPermissionId() {
 		return this.getId();
 	}
+
 	public String getOwner() {
-		
 		return null;
 	}
-	public void setOwner(String owner) {
 
-		
+	public void setOwner(String owner) {
+		// Not implemented
 	}
-	
-	
+
 	@JsonIgnore
 	@Override
 	public String toString() {
 		return "WorkflowAction [id=" + id + ", name=" + name + ", stepId=" + stepId + ", nextStep=" + nextStep + "]";
 	}
+
 	@JsonIgnore
 	public List<PermissionSummary> acceptedPermissions() {
 		List<PermissionSummary> accepted = new ArrayList<PermissionSummary>();
 		accepted.add(new PermissionSummary("use",
 				"use-permission-description", PermissionAPI.PERMISSION_USE));
 		return accepted;
 	}
+
 	@JsonIgnore
 	public List<RelatedPermissionableGroup> permissionDependencies(int requiredPermission) {
 		return null;
 	}
+
 	@JsonIgnore
 	public Permissionable getParentPermissionable() throws DotDataException {
 		return null;
 	}
+
 	@JsonIgnore
 	public String getPermissionType() {
 		return this.getClass().getCanonicalName();
 	}
+
 	@JsonIgnore
 	public boolean isParentPermissionable() {
 		return false;
 	}
-	private static final long serialVersionUID = 1L;
 
-	String id;
-	String name;
-	String stepId;
-	String condition;
-	String nextStep;
-	String nextAssign;
-	String icon;
-	boolean roleHierarchyForAssign;
-	boolean requiresCheckout;
-	
+	/**
+	 * @deprecated see {@link #getRequiresCheckoutOption()}
+	 * @return boolean
+	 */
 	public boolean requiresCheckout() {
 		return requiresCheckout;
 	}
+
+	/**
+	 * @deprecated see {@link #getRequiresCheckoutOption()}
+	 * @return boolean
+	 */
+	@Deprecated
 	public boolean isRequiresCheckout() {
 		return requiresCheckout;
     }
+
+	/**
+	 * @deprecated see {@link #setRequiresCheckoutOption(String)}
+	 * p-pp0()}
+	 * @param requiresCheckout
+	 */
+	@Deprecated
 	public void setRequiresCheckout(boolean requiresCheckout) {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)